mozilla/bedrock
# Changes needed to support SSO for admin access on Demos

## One-line summary

This changeset contains tweaks to Bedrock that enable us to bootstrap SSO for CMS/Admin access on Demo services.

## Significant changes and points to review

This changeset is needed to make the updates to https://github.com/mozmeao/infra-services/pull/184 work. 

It's centered around passing in an env var that specs the SSO users to set up (via the changes already merged in https://github.com/mozilla/bedrock/pull/15436)

It also contains a significant new behaviour: the option to deliberately _not_ download a fresh database during the `run-file-clock.sh` script, as getting a new DB would throw away the DB state that contains the SSO users (and and Switches or CMS content amended on that demo box). 

(It's made optional so that it doesn't change the bedrock-test deployment's behaviour, but I think we're headed down a route where we can remove the script entirely now we're using Postgres in production.)

## Issue / Bugzilla link

Resolves #15375 

## Testing

At the moment Demo 5 has the SSO users bootstrapped for the entire Websites team. It also uses the same people.m.o group as Bedrock Dev, so if you have access to the CMS in the Dev deployment you should have access to a Demo. 

* Head to https://www-demo5.allizom.org/cms-admin/
* Sign in with SSO
* Confirm you can add and publish a new page in the CMS
* Use the sidebar link to go to the Django admin
* Confirm you can amend a switch (the DE homepage one is easy to see in effect if go /de/ before and after)

## Caveats

* Cloud Run services go to sleep after 15 mins unless they are deliberately kept awake - see "Always on vs auto-sleep demo servers" on https://bedrock.readthedocs.io/en/latest/contribute.html#demo-sites. I think the way things are set up now will avoid data loss, but if we see that switch settings are being lost because a cold start involves spinning up a whole new container (and therefore using an earlier state of the sqlite DB), we can look to retain demos as 'always on'

* Demos use a sqlite copy of the Production DB. Images are not synced over from the prod CMS bucket and I don't really want to hook up the demos to the prod bucket. This means CMS-backed images are not currently available in demos - whether we need to support that in practical terms is TBC, so we'll deal with that if/when we need it.
